src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -594,6 +613,42 @@ private async void ReconnectLoop()
         }
     }
 
+    private async Task PublishEventsAsync()
+    {
+        try
+        {
+            while (!_disposedCancellationTokenSource.IsCancellationRequested)
+            {
+                await _eventChannel.Reader.WaitToReadAsync().ConfigureAwait(false);
+                if (_eventChannel.Reader.TryRead(out var eventData))
```

Comment:
Should we not check return value of `WaitToReadAsync()` and use while loop for `TryRead()`?

Reply:
Good point a short comment would be nice.

Reply:
There is technically already a while loop, and since this is only for event notifications I don't think it's worth adding an inner while loop to empty the queue. But using the return value makes sense, I will update the PR.